### ISSUE TITLE
Warn when 'add item' will override an existing value?

### ISSUE THREAD

**Issue by mjbvz** [microsoft/vscode#104130]
Issue Type: <b>Bug</b>

**Repro**
1. Using the `files.associations` setting
1. Add a key `abc` in the setting UI
1. Now click add item again and enter the key `abc` and a different value

**Bug**
This silently overwrites the previous value. Perhaps we should show a warning in this case 

VS Code version: Code - Insiders 1.48.0-insider (d13d2fc56da7a2f8bcad4256212db0661fcbba45, 2020-08-05T05:26:44.946Z)
OS version: Darwin x64 19.6.0

<details>
<summary>System Info</summary>

|Item|Value|
|---|---|
|CPUs|Intel(R) Core(TM) i9-9980HK CPU @ 2.40GHz (16 x 2400)|
|GPU Status|2d_canvas: enabled<br>flash_3d: enabled<br>flash_stage3d: enabled<br>flash_stage3d_baseline: enabled<br>gpu_compositing: enabled<br>metal: disabled_off<br>multiple_raster_threads: enabled_on<br>oop_rasterization: disabled_off<br>protected_video_decode: unavailable_off<br>rasterization: enabled<br>skia_renderer: disabled_off_ok<br>video_decode: enabled<br>viz_display_compositor: enabled_on<br>viz_hit_test_surface_layer: disabled_off_ok<br>webgl: enabled<br>webgl2: enabled|
|Load (avg)|2, 2, 2|
|Memory (System)|32.00GB (0.15GB free)|
|Process Argv||
|Screen Reader|no|
|VM|14%|
</details>

**Comment by 9at8** [comment]
Something that looks like a validation?

![image](https://user-images.githubusercontent.com/8235156/89475151-6f6fa080-d755-11ea-9259-75dd7352fcf8.png)

**Comment by mjbvz** [comment]
Yes that sounds good. With `files.associations`, I just wanted to be notified that I was overriding an existing value
